steps/expval.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The PennyLane-Orquestra step for computing the expectation value of Hermitian
+operators given a quantum circuit and a quantum device.
+
+Functions defined in this file may be included in an Orquestra workflow file as
+a workflow step. Such workflow steps are executed on a remote Orquestra node.
+"""
+import json
+
+import numpy as np
+from openfermion import IsingOperator, QubitOperator
+from qiskit import QuantumCircuit
+
+from zquantum.core.circuit import Circuit
+from zquantum.core.measurement import expectation_values_to_real
+from zquantum.core.utils import create_object, save_list
+
+
+def run_circuit_and_get_expval(
+    backend_specs: dict,
+    circuit: str,
+    operators: str,
+):
+    """Takes a circuit to obtain the expectation value of an operator on a
+    given backend.
+
+    All backend calls used in this function are defined as ``QuantumBackend``
+    and ``QuantumSimulator`` interface standard methods implemented are defined
+    in the ``z-quantum-core`` repository.
```

Comment:
This sentence was a bit difficult to parse for me; it feels a bit wrong. :confused:

Reply:
Oh, good point! Rephrased it.